pkg/grapicmd/context.go

```diff
@@ -45,51 +36,67 @@ type Config struct {
 	}
 }
 
+type BuildConfig struct {
+	AppName   string
+	Version   string
+	Revision  string
+	BuildDate string
+	Prebuilt  bool
+}
+
 // Init initializes the runtime context.
-func (c *Ctx) Init() {
+func (c *Ctx) Init() error {
 	if c.FS == nil {
 		c.FS = afero.NewOsFs()
 	}
 
 	if c.Viper == nil {
 		c.Viper = viper.New()
-		c.Viper.SetFs(c.FS)
 	}
 
+	c.Viper.SetFs(c.FS)
+
 	if c.Execer == nil {
 		c.Execer = exec.New()
 	}
 
-	if c.RootDir == "" {
-		c.RootDir, c.InsideApp = fs.LookupRoot(c.FS, c.CurrentDir)
+	if c.Build.AppName == "" {
+		c.Build.AppName = "grapi"
 	}
 
-	if c.InsideApp && c.BinDir == "" {
-		c.BinDir = filepath.Join(c.RootDir, "bin")
-	}
+	return errors.WithStack(c.loadConfig())
 }
 
-// Load reads configurations from the config file.
-func (c *Ctx) Load(cfgFile string) error {
-	if !c.InsideApp {
-		return nil
+func (c *Ctx) loadConfig() error {
+	c.Viper.SetConfigName(c.Build.AppName)
+	for dir := c.RootDir.String(); dir != "/"; dir = filepath.Dir(dir) {
+		c.Viper.AddConfigPath(dir)
 	}
 
-	c.Viper.SetConfigFile(cfgFile)
 	err := c.Viper.ReadInConfig()
 	if err != nil {
-		return errors.WithStack(err)
+		clog.Info("failed to find config file", "error", err)
+		return nil
 	}
 
-	err = c.Viper.UnmarshalKey("config", &c.Config)
+	c.insideApp = true
+	c.RootDir = cli.RootDir(filepath.Dir(c.Viper.ConfigFileUsed()))
+
+	err = c.Viper.Unmarshal(&c.Config)
 	if err != nil {
+		clog.Warn("failed to parse config", "error", err)
 		return errors.WithStack(err)
 	}
 
 	err = c.Viper.UnmarshalKey("protoc", &c.ProtocConfig)
 	if err != nil {
+		clog.Warn("failed to parse protoc config", "error", err)
 		return errors.WithStack(err)
 	}
 
 	return nil
 }
+
+func (c *Ctx) IsInsideApp() bool {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method Ctx.IsInsideApp should have comment or be unexported

Reply:
fixed in 3041c36